docs/dev_guide/external_objectives.rst

```diff
@@ -0,0 +1,316 @@
+====================
+Using external codes
+====================
+
+The ``ExternalObjective`` class allows external codes to be used within a DESC
+optimization. This takes a user-supplied function that does not need to be JAX
+transformable, and wraps it with forward finite differences. This is most useful for
+cases when the function needs to call external codes that are written in another
+language besides Python, or cannot be rewritten with JAX for some other reason. If the
+user function can be made JAX transformable, it is recommended to use the
+``ObjectiveFromUser`` class instead.
+
+This guide begins with a simple example of how to use a custom function with the
+``ExternalObjective``. It then uses the ``TERPSICHORE`` objective as a more realistic
+example, which can be used as a template for wrapping other codes. Finally, an example
+script is shown for how to run the TERPSICHORE optimization with multiprocessing. (You
+must obtain access to the TERPSICHORE source code to run this example, it is not
+included with DESC.)
+
+Simple example
+--------------
+
+The following is a simple example of a custom function that is not JAX transformable.
+It was adapted from the test ``test_external_vs_generic_objectives`` in
+``tests/test_examples.py``.
+
+The function must take a single positional argument, which can be either a single
```

Comment:
specify that this is `fun`, and that this requirement is a requirement of `ExternalObjective`, not just a requirement of the specific function you chose for this example